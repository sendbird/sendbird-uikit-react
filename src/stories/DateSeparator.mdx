import { Meta, Story, Canvas, Controls } from '@storybook/addon-docs';
import DateSeparator from '../ui/DateSeparator';
import * as DateSeparatorStories from './DateSeparator.stories';

<Meta title="UI/DateSeparator" component={DateSeparator} of={DateSeparatorStories} />

# DateSeparator

The `DateSeparator` component is used to visually separate content by dates. It can display a custom text or use a default text label.

<Canvas of={DateSeparatorStories.Default} />
<Controls />

### Importing DateSeparator

To integrate the DateSeparator component into your application, use the following import statement:

```tsx
import DateSeparator from '@sendbird/uikit-react/ui/DateSeparator';
```

### Examples

#### Default
Displays the default DateSeparator with predefined styles.
<Canvas of={DateSeparatorStories.Default} />

#### Custom Color
Illustrates the DateSeparator with a custom color applied.
<Canvas of={DateSeparatorStories.CustomColor} />
