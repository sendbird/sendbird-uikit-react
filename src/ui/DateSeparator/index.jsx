import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';
import { changeColorToClassName, Colors } from '../../utils/color';
import Label, { LabelTypography, LabelColors } from '../Label';

export default function DateSeparator({
  className,
  children,
  separatorColor,
}) {
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-separator',
      ].join(' ')}
    >
      <div className={['sendbird-separator__left', `${changeColorToClassName(separatorColor)}--background-color`].join(' ')} />
      <div className="sendbird-separator__text">{children}</div>
      <div className={['sendbird-separator__right', `${changeColorToClassName(separatorColor)}--background-color`].join(' ')} />
    </div>
  );
}

DateSeparator.propTypes = {
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
    PropTypes.element,
  ]),
  separatorColor: PropTypes.oneOf([
    'ONBACKGROUND_1',
    'ONBACKGROUND_2',
    'ONBACKGROUND_3',
    'ONBACKGROUND_4',
    'ONCONTENT_1',
    'PRIMARY',
    'ERROR',
  ]),
};

DateSeparator.defaultProps = {
  className: '',
  children: (
    <Label type={LabelTypography.CAPTION_2} color={LabelColors.ONBACKGROUND_2}>
      Date Separator
    </Label>
  ),
  separatorColor: Colors.ONBACKGROUND_4,
};
