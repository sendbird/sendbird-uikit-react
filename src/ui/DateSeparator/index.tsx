import React, { ReactElement } from 'react';

import './index.scss';

import {
  Colors,
  changeColorToClassName,
} from '../../utils/color';

import Label, { LabelTypography, LabelColors } from '../Label';

export interface DateSeparatorProps {
  children?: string | ReactElement;
  className?: string | Array<string>;
  separatorColor?: Colors;
}
const DateSeparator = ({
  children = null,
  className = '',
  separatorColor = Colors.ONBACKGROUND_4,
}: DateSeparatorProps): ReactElement => {
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-separator',
      ].join(' ')}
    >
      <div className={['sendbird-separator__left', `${changeColorToClassName(separatorColor)}--background-color`].join(' ')} />
      <div className="sendbird-separator__text">
        {
          children
          || (
            <Label type={LabelTypography.CAPTION_2} color={LabelColors.ONBACKGROUND_2}>
              Date Separator
            </Label>
          )
        }
      </div>
      <div className={['sendbird-separator__right', `${changeColorToClassName(separatorColor)}--background-color`].join(' ')} />
    </div>
  );
};

export default DateSeparator;
