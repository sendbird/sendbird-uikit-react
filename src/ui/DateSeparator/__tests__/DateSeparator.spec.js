import React from 'react';
import { render } from '@testing-library/react';

import DateSeparator from "../index";

describe('ui/DateSeparator', () => {
  it('should contain className', function () {
    const className = "example-classname";
    const { container } = render(<DateSeparator className={className} />);

    expect(container.getElementsByClassName('sendbird-separator')[0].className).toContain(className);
    expect(container.getElementsByClassName('sendbird-separator')[0].className).toContain('sendbird-separator');
  });

  it('should do a snapshot test of the DateSeparator DOM', function () {
    const className = "example-classname";
    const { asFragment } = render(<DateSeparator className={className} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
