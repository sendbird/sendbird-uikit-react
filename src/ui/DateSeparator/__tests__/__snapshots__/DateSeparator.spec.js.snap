// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/DateSeparator should do a snapshot test of the DateSeparator DOM 1`] = `
<DocumentFragment>
  <div
    class="example-classname sendbird-separator"
  >
    <div
      class="sendbird-separator__left sendbird-color--onbackground-4--background-color"
    />
    <div
      class="sendbird-separator__text"
    >
      <span
        class="sendbird-label sendbird-label--caption-2 sendbird-label--color-onbackground-2"
      >
        Date Separator
      </span>
    </div>
    <div
      class="sendbird-separator__right sendbird-color--onbackground-4--background-color"
    />
  </div>
</DocumentFragment>
`;
